### PR TITLE
Load wasm builds if available

### PR DESCRIPTION
**Pull request by LianaHus**
fix https://github.com/ethereum/remix-ide/issues/2822

### CODE REVIEW

File: src/app/tabs/compileTab/compilerContainer.js [hunk 1]
```diff
@@ -25,7 +24,8 @@ class CompilerContainer {
       allversions: null,
       selectedVersion: null,
       defaultVersion: 'soljson-v0.6.6+commit.6c089d02.js', // this default version is defined: in makeMockCompiler (for browser test) and in package.json (downloadsolc_root) for the builtin compiler
-      baseurl: 'https://solc-bin.ethereum.org/bin'
+      baseURLWasm: baseURLWasm,
```
Comment (yann300) [inline src/app/tabs/compileTab/compilerContainer.js:27]: you don't need to make it a data property, you are importing it

File: src/app/tabs/compileTab/compilerContainer.js [hunk 2]
```diff
@@ -413,27 +413,42 @@ class CompilerContainer {
...
-          addTooltip('Cannot load compiler version list. It might have been blocked by an advertisement blocker. Please try deactivating any of them from this page and reload.')
-        }
-      } else {
-        allversions = [{ path: 'builtin', longVersion: 'latest local version' }]
-        selectedVersion = 'builtin'
+  // fetching both normal and wasm builds and creating a [version, baseUrl] map
+  async fetchAllVersion (callback) {
+    let allVersions, selectedVersion, allVersionsWasm
+    // fetch normal builds
+    const binRes = await promisedMiniXhr(`${this.data.baseURLBin}/list.json`)
```
Comment (yann300) [inline src/app/tabs/compileTab/compilerContainer.js:420]: `const binRes = await promisedMiniXhr(`${baseURLBin}/list.json`)` instead?

**Comment by yann300** [comment]
do we need new e2e tests or is this already tested?

**Comment by LianaHus** [comment]
> do we need new e2e tests or is this already tested?

I've added some. Please take a look at diff, Feel free to share more ideas on how to test it if you have any.

File: test-browser/commands/noWorkerErrorFor.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    })
+    return this
+  }
+}
+
+function noWorkerErrorFor (browser, version, done) {
+  browser
+    .clickLaunchIcon('solidity')
+    .setSolidityCompilerVersion(version)
+    .waitForElementPresent('*[data-id="compiledErrors"]')
```
Comment (yann300) [inline test-browser/commands/noWorkerErrorFor.js:19]: is everything recompiled after version is  changed?

File: package.json [hunk 4]
```diff
@@ -81,6 +81,7 @@
   "dependencies": {
     "@remixproject/engine": "^0.2.3",
     "http-server": "^0.11.1",
+    "selenium": "^2.20.0",
```
Comment (yann300) [inline package.json:84]: not sure you need that here

Comment (LianaHus) [reply package.json:84]: Something went wrong while I was trying to test locally :).  Will remove this. Thanks

File: test-browser/commands/noWorkerErrorFor.js (hunk 3 above)
Comment (LianaHus) [reply test-browser/commands/noWorkerErrorFor.js:19]: any ideas on how I can check that besides waiting ?

Comment (yann300) [reply test-browser/commands/noWorkerErrorFor.js:19]: if you run it locally you'll see it

Comment (LianaHus) [reply test-browser/commands/noWorkerErrorFor.js:19]: yes but is there a way to assert it?

**Pull request merged**